### PR TITLE
Adds count and count_recursive methods for Tree objects.

### PR DESCRIPTION
**Pull request by bartkamphorst**

### CODE REVIEW

File: lib/tree.rb [hunk 1]
```diff
@@ -30,6 +30,14 @@ def contents_array
       @contents_ary ||= jtree_entries
     end
     
+    def recursive_count(limit = nil)
+      jtree_entries({recursive: true, limit: limit}).size
```
Comment (dometto) [inline lib/tree.rb]: Would it make sense to cache the results of this operation? Or not, because of the possible different `limits`?

**Comment by dometto** [comment]
:+1:

File: lib/tree.rb (hunk 1 above)
Comment (bartkamphorst) [inline lib/tree.rb]: Yes, that's what I was thinking. Not sure if it's worth the trouble, but I suppose we could cache the limit as well, and rerun only if the limit changes?

Comment (dometto) [inline lib/tree.rb]: Caching the different limits is very cheap memory-wise (a hash with integer keys and values), while the computation can be quite expensive. On the other hand, it seems likely that users would store the result in a variable themselves, so maybe it's too much. Dunno. :)

Comment (bartkamphorst) [inline lib/tree.rb]: What about this?

**Comment by bartkamphorst** [comment]
Any thoughts on the latest proposal?

**Comment by dometto** [comment]
Looks fine!

**Pull request merged**